// vitepress-plugin-pagefind/dist/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  chineseSearchOptimize: () => chineseSearchOptimize,
  pagefindPlugin: () => pagefindPlugin
});
module.exports = __toCommonJS(src_exports);
var import_node_path2 = __toESM(require("path"));
var import_node_url = require("url");
var import_node_process2 = __toESM(require("process"));
var import_javascript_stringify = require("javascript-stringify");

// src/node.ts
var import_node_fs = __toESM(require("fs"));
var import_node_child_process = require("child_process");
var import_node_path = __toESM(require("path"));
var import_node_os = __toESM(require("os"));
var import_node_process = __toESM(require("process"));
var import_gray_matter = __toESM(require("gray-matter"));
var import_fast_glob = __toESM(require("fast-glob"));

// src/utils/index.ts
function formatDate(d, fmt = "yyyy-MM-dd hh:mm:ss") {
  if (!(d instanceof Date)) {
    d = new Date(d);
  }
  const o = {
    "M+": d.getMonth() + 1,
    // 月份
    "d+": d.getDate(),
    // 日
    "h+": d.getHours(),
    // 小时
    "m+": d.getMinutes(),
    // 分
    "s+": d.getSeconds(),
    // 秒
    "q+": Math.floor((d.getMonth() + 3) / 3),
    // 季度
    "S": d.getMilliseconds()
    // 毫秒
  };
  if (/(y+)/.test(fmt)) {
    fmt = fmt.replace(
      RegExp.$1,
      `${d.getFullYear()}`.substr(4 - RegExp.$1.length)
    );
  }
  for (const k in o) {
    if (new RegExp(`(${k})`).test(fmt))
      fmt = fmt.replace(
        RegExp.$1,
        RegExp.$1.length === 1 ? o[k] : `00${o[k]}`.substr(`${o[k]}`.length)
      );
  }
  return fmt;
}

// src/node.ts
function getPagesData(srcDir, config, pluginSiteConfig2) {
  const files = import_fast_glob.default.sync(`${srcDir}/**/*.md`, { ignore: ["node_modules"] });
  return files.map((file) => {
    const route = config.site.base + normalizePath(import_node_path.default.relative(config.srcDir, file)).replace(/(^|\/)index\.md$/, "$1").replace(/\.md$/, config.cleanUrls ? "" : ".html");
    const fileContent = import_node_fs.default.readFileSync(file, "utf-8");
    const { data: frontmatter, excerpt } = (0, import_gray_matter.default)(fileContent, {
      excerpt: true
    });
    const meta = {
      description: excerpt,
      ...frontmatter
    };
    if (!meta.title) {
      meta.title = getDefaultTitle(fileContent);
    }
    if (!meta.date) {
      meta.date = getFileBirthTime(file);
    } else {
      const timeZone = pluginSiteConfig2?.timeZone ?? 8;
      meta.date = formatDate(
        /* @__PURE__ */ new Date(`${new Date(meta.date).toUTCString()}+${timeZone}`)
      );
    }
    return {
      route,
      meta
    };
  }).filter((v) => v.meta.layout !== "home");
}
var windowsSlashRE = /\\/g;
var isWindows = import_node_os.default.platform() === "win32";
function slash(p) {
  return p.replace(windowsSlashRE, "/");
}
function normalizePath(id) {
  return import_node_path.default.posix.normalize(isWindows ? slash(id) : id);
}
function getDefaultTitle(content) {
  const title = clearMatterContent(content).split("\n")?.find((str) => {
    return str.startsWith("# ");
  })?.slice(2).replace(/[\s]/g, "") || "";
  return title;
}
function clearMatterContent(content) {
  let first___;
  let second___;
  const lines = content.split("\n").reduce((pre, line) => {
    if (!line.trim() && pre.length === 0) {
      return pre;
    }
    if (line.trim() === "---") {
      if (first___ === void 0) {
        first___ = pre.length;
      } else if (second___ === void 0) {
        second___ = pre.length;
      }
    }
    pre.push(line);
    return pre;
  }, []);
  return lines.slice(second___ || 0).join("\n");
}
function getFileBirthTime(url) {
  let date = /* @__PURE__ */ new Date();
  try {
    const infoStr = (0, import_node_child_process.spawnSync)("git", ["log", "-1", '--pretty="%ci"', url]).stdout?.toString().replace(/["']/g, "").trim();
    if (infoStr) {
      date = new Date(infoStr);
    }
  } catch (error) {
    return formatDate(date);
  }
  return formatDate(date);
}
var ignoreSelectors = [
  // 侧边栏内容
  "div.aside",
  // 标题锚点
  "a.header-anchor"
];
var EXTERNAL_URL_RE = /^[a-z]+:/i;
function joinPath(base, path3) {
  return `${base}${path3}`.replace(/\/+/g, "/");
}
function withBase(base, path3) {
  return EXTERNAL_URL_RE.test(path3) || path3.startsWith(".") ? path3 : joinPath(base, path3);
}
var pluginSiteConfig = {
  /**
   * TODO：支持更多pagefind配置项
   * vitepress buildEnd钩子调用
   */
  buildEnd(ctx) {
    const pagefindOps = ctx.PagefindOption;
    const ignore = [
      ...new Set(ignoreSelectors.concat(pagefindOps?.excludeSelector || []))
    ];
    const { log } = console;
    log();
    log("=== pagefind: https://pagefind.app/ ===");
    const siteDir = import_node_path.default.join(
      import_node_process.default.argv.slice(2)?.[1] || ".",
      ".vitepress/dist"
    );
    let command = `npx pagefind --site ${siteDir}`;
    if (ignore.length) {
      command += ` --exclude-selectors "${ignore.join(", ")}"`;
    }
    if (typeof pagefindOps.forceLanguage === "string") {
      command += ` --force-language ${pagefindOps.forceLanguage}`;
    }
    if (pagefindOps.indexingCommand) {
      command = pagefindOps.indexingCommand;
    }
    log(command);
    log();
    (0, import_node_child_process.execSync)(command, {
      stdio: "inherit"
    });
  },
  transformHead(ctx) {
    return [
      [
        "script",
        {},
        `import('${withBase(ctx.siteData.base || "", "/pagefind/pagefind.js")}')
    .then((module) => {
      window.__pagefind__ = module
      module.init()
    })
    .catch(() => {
      // console.log('not load /pagefind/pagefind.js')
    })`
      ]
    ];
  }
};
function chineseSearchOptimize(input) {
  return input.replace(/[\u4E00-\u9FA5]/g, " $& ").replace(/\s+/g, " ").trim();
}

// src/index.ts
var import_meta = {};
function isESM() {
  return typeof __filename === "undefined" || typeof __dirname === "undefined";
}
function getDirname() {
  return isESM() ? import_node_path2.default.dirname((0, import_node_url.fileURLToPath)(import_meta.url)) : __dirname;
}
var aliasSearchVueFile = `${getDirname()}/../src/Search.vue`;
function pagefindPlugin(searchConfig = {}) {
  const virtualModuleId = "virtual:pagefind";
  const resolvedVirtualModuleId = `\0${virtualModuleId}`;
  let resolveConfig;
  const pluginOps = {
    name: "vitepress-plugin-pagefind",
    enforce: "pre",
    config: () => ({
      resolve: {
        alias: {
          "./VPNavBarSearch.vue": aliasSearchVueFile
        }
      }
    }),
    configResolved(config) {
      if (resolveConfig) {
        return;
      }
      resolveConfig = config;
      const vitepressConfig = config.vitepress;
      if (!vitepressConfig) {
        return;
      }
      const selfBuildEnd = vitepressConfig.buildEnd;
      vitepressConfig.buildEnd = (siteConfig) => {
        selfBuildEnd?.(siteConfig);
        siteConfig = Object.assign(siteConfig || {}, {
          PagefindOption: searchConfig
        });
        pluginSiteConfig?.buildEnd?.(siteConfig);
      };
      const selfTransformHead = vitepressConfig.transformHead;
      vitepressConfig.transformHead = async (ctx) => {
        const selfHead = await Promise.resolve(selfTransformHead?.(ctx)) || [];
        const pluginHead = await Promise.resolve(pluginSiteConfig?.transformHead?.(ctx)) || [];
        return selfHead.concat(pluginHead);
      };
    },
    async resolveId(id) {
      if (id === virtualModuleId) {
        return resolvedVirtualModuleId;
      }
    },
    // 文章数据
    load(id) {
      if (id !== resolvedVirtualModuleId)
        return;
      const srcDir = resolveConfig.vitepress.srcDir.replace(resolveConfig.vitepress.root, "").replace(/^\//, "") || import_node_process2.default.argv.slice(2)?.[1] || ".";
      const docsData = getPagesData(
        srcDir,
        resolveConfig.vitepress,
        searchConfig
      );
      return `
      import { ref } from 'vue'
      export const docs = ref(${JSON.stringify(docsData)})
      export const searchConfig = ${(0, import_javascript_stringify.stringify)(searchConfig)}
      `;
    },
    // 添加检索的内容标识
    transform(code, id) {
      if (id.endsWith("theme-default/Layout.vue")) {
        return code.replace("<VPContent>", "<VPContent data-pagefind-body>");
      }
      return code;
    }
  };
  return pluginOps;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  chineseSearchOptimize,
  pagefindPlugin
});
